src/components/screens/new-questionnaire-screen.tsx

```diff
@@ -7,48 +22,90 @@ import { KeyboardReturn } from "@mui/icons-material";
 import strings from "src/localization/strings";
+import { useLambdasApi } from "src/hooks/use-api";
 
 /**
  * New Questionnaire Screen component
- * Includes a card (new-questionnaire-card.tsx) to create questions for a new questionnaire
+ * Includes a card (new-questionnaire-card.tsx) to create options for a new questionnaire
  */
 const NewQuestionnaireScreen = () => {
   const adminMode = UserRoleUtils.adminMode();
-  
+  const { questionnaireApi } = useLambdasApi();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
   const [questionnaireTitle, setQuestionnaireTitle] = useState("");
+  const [questionnaireDescription, setQuestionnaireDescription] = useState("");
+  const [options, setOptions] = useState<Question[]>([]);
+  const [passScoreValue, setPassScoreValue] = useState(0);
 
   const handleQuestionTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
     setQuestionnaireTitle(event.target.value);
   };
 
-  /**
-   * State and functions to handle the questions and options
-   */
-  const [questions, setQuestions] = useState<Question[]>([]);
+  const handleDescriptionChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuestionnaireDescription(event.target.value);
+  };
+
+  const handlePassScoreSliderChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setPassScoreValue(Number(event.target.value));
+  };
 
   /**
    * Functions to add new question to Questionnaire
-   * @param questionText 
-   * @param options 
+   * @param question
+   * @param options
    */
-  const handleAddQuestionSubmit = (
-    questionText: string,
-    options: QuestionOption[]
-  ) => {
-    setQuestions((prevQuestions: Question[]) => [...prevQuestions, { questionText, options }]);
+  const handleAddQuestionSubmit = (question: string, options: QuestionOption[]) => {
+    setOptions((prevQuestions: Question[]) => [...prevQuestions, { question, options }]);
   };
 
   const handleDeleteQuestion = (index: number) => {
-    const updatedQuestions = [...questions];
+    const updatedQuestions = [...options];
     updatedQuestions.splice(index, 1);
-    setQuestions(updatedQuestions);
+    setOptions(updatedQuestions);
+  };
+
+  /**
+   * Function to count all correct answers in the questionnaire, used for passScore determination
+   */
+  const countCorrectAnswers = () => {
+    return options.reduce((count, question) => {
+      return count + question.options.filter((option) => option.value === true).length;
+    }, 0);
   };
 
   /**
-   * This should save the question title and questions to DB.
-   * TODO:
+   * Function to save the new questionnaire
    */
-  const handleSaveSubmit = () => {};
+  const handleSaveSubmit = async () => {
+    try {
+      setLoading(true);
+      setError(null);
+
+      const createdQuestionnaire = await questionnaireApi.createQuestionnaire({
+        questionnaire: {
+          id: "",
```

Comment:
Do we have to pass an empty string here? I would hope the lambda would assign the id, but I would also hope we don't need to provide id as empty string at this point, if we do we might need to update the specs to the type of id is optional?

Reply:
Bugfix helped